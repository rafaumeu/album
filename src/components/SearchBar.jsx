import React from 'react'

const SearchBar = () => {
  const categorias = ['Natureza', 'Pessoas', 'Animais', 'Esportes']
  return (
    <div className='search-bar'>
      <input type='text' placeholder='Pesquisar fotos...' />
      <button>Pesquisar</button>
      <select>
        {categorias.map((categoria) => (
          <option value={categoria} key={categoria}>
            {categoria}
          </option>
        ))}
      </select>
    </div>
  )
}

export default SearchBar
