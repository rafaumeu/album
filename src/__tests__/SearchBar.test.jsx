import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { mount } from 'vitest'

import SearchBar from '../components/SearchBar'

describe('<SearchBar />', () => {
  it('should renders without crashing', () => {
    const { debug, container } = render(
      <SearchBar
        setActivateSearch={() => {}}
        setCategoria={() => {}}
        setQuery={() => {}}
      />
    )
    const inputElement = screen.getByPlaceholderText('Pesquisar fotos...')
    const buttonElement = screen.getByText(/pesquisar/i)
    const selectElement = container.querySelector('select')
    expect(inputElement).toBeInTheDocument()
    expect(buttonElement).toBeInTheDocument()
    expect(selectElement).toBeInTheDocument()
  })

  it('should trigger callback functions on user interaction', async () => {
    const setQueryMock = vi.fn()
    const setActivateSearchMock = vi.fn()
    const setCategoriasMock = vi.fn()
    render(
      <SearchBar
        setQuery={setQueryMock}
        setActivateSearch={setActivateSearchMock}
        setCategoria={setCategoriasMock}
      />
    )

    const inputElement = screen.getByPlaceholderText('Pesquisar fotos...')
    const buttonElement = screen.getByText(/pesquisar/i)

    // Simula a mudança no valor do input
    fireEvent.change(inputElement, { target: { value: 'pesquisa' } })

    // Simula o clique no botão
    fireEvent.click(buttonElement)

    // Verifica se o estado local (localQuery) foi atualizado corretamente
    expect(inputElement.value).toBe('pesquisa')
  })
  it('should trigger callback functions on user interaction', async () => {
    const setQueryMock = vi.fn()
    const setActivateSearchMock = vi.fn()
    const setCategoriasMock = vi.fn()

    const { container } = render(
      <SearchBar
        setQuery={setQueryMock}
        setActivateSearch={setActivateSearchMock}
        setCategoria={setCategoriasMock}
      />
    )

    const inputElement = screen.getByPlaceholderText('Pesquisar fotos...')
    const buttonElement = screen.getByText(/pesquisar/i)
    const selectElement = container.querySelector('select') // ou ajuste conforme seu código

    // Simula a mudança no valor do input
    fireEvent.change(inputElement, { target: { value: 'pesquisa' } })

    // Simula o clique no botão
    fireEvent.click(buttonElement)

    // Aguarda a execução das funções assíncronas

    expect(inputElement.value).toBe('pesquisa')

    // Simula a mudança na seleção da categoria
    fireEvent.change(selectElement, { target: { value: 'Natureza' } })

    // Aguarda a execução das funções assíncronas após a mudança na categoria
    await waitFor(() => {
      // Verifica se setCategoria foi chamado corretamente
      expect(setCategoriasMock).toHaveBeenCalledWith('Natureza')
      expect(setActivateSearchMock).toHaveBeenCalledWith(true)
    })
  })
  it('should to be matches snapshot', async () => {
    const setQueryMock = vi.fn()
    const setActivateSearchMock = vi.fn()
    const setCategoriasMock = vi.fn()

    const { container, debug } = render(
      <SearchBar
        setQuery={setQueryMock}
        setActivateSearch={setActivateSearchMock}
        setCategoria={setCategoriasMock}
      />
    )
    const inputElement = screen.getByPlaceholderText('Pesquisar fotos...')
    const buttonElement = screen.getByText(/pesquisar/i)
    const selectElement = container.querySelector('select') // ou ajuste conforme seu código

    // Simula a mudança no valor do input
    fireEvent.change(inputElement, { target: { value: 'pesquisa' } })

    // Simula o clique no botão
    fireEvent.click(buttonElement)

    // Aguarda a execução das funções assíncronas

    expect(inputElement.value).toBe('pesquisa')

    // Simula a mudança na seleção da categoria
    fireEvent.change(selectElement, { target: { value: 'Natureza' } })
    expect(container).toMatchSnapshot()
    screen.debug()
  })
})
