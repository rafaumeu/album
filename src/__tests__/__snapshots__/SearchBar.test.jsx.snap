// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SearchBar /> > should to be matches snapshot 1`] = `
<div>
  <div
    class="search-bar"
  >
    <input
      placeholder="Pesquisar fotos..."
      type="text"
      value="pesquisa"
    />
    <button>
      Pesquisar
    </button>
    <select>
      <option
        value=""
      >
        Todas as categorias
      </option>
      <option
        value="Natureza"
      >
        Natureza
      </option>
      <option
        value="Pessoas"
      >
        Pessoas
      </option>
      <option
        value="Animais"
      >
        Animais
      </option>
      <option
        value="Esportes"
      >
        Esportes
      </option>
      <option
        value="Tecnologia"
      >
        Tecnologia
      </option>
    </select>
  </div>
</div>
`;
